chain/src/chain.rs

```diff
@@ -1693,47 +1712,26 @@ impl BlockChain {
     }
 
     pub fn get_block_dag_origin(&self) -> Result<HashValue> {
-        let dag_genesis = self.get_block_dag_genesis(&self.current_header())?;
+        let dag_genesis = self.get_block_dag_genesis()?;
         let block_header = self
             .storage
             .get_block_header_by_hash(dag_genesis)?
             .ok_or_else(|| anyhow!("Cannot find block by hash {:?}", dag_genesis))?;
-
-        Ok(HashValue::sha3_256_of(
-            &[block_header.parent_hash(), block_header.id()].encode()?,
-        ))
+        Ok(block_header.parent_hash())
     }
 
-    pub fn get_dag_state_by_block(&self, header: &BlockHeader) -> Result<(HashValue, DagState)> {
-        let dag_genesis = self.get_block_dag_genesis(header)?;
+    pub fn get_dag_state_by_block(&self) -> Result<(HashValue, DagState)> {
+        let dag_genesis = self.get_block_dag_genesis()?;
         Ok((dag_genesis, self.dag.get_dag_state(dag_genesis)?))
     }
 
-    pub fn check_dag_type(&self, header: &BlockHeader) -> Result<DagHeaderType> {
-        use std::cmp::Ordering;
-
-        let dag_height = self.dag_fork_height()?.unwrap_or(u64::MAX);
-        if header.is_genesis() {
-            return Ok(DagHeaderType::Single);
-        }
-        let no_parents = header.parents_hash().unwrap_or_default().is_empty();
-
-        match (no_parents, header.number().cmp(&dag_height)) {
-            (true, Ordering::Greater) => {
-                Err(anyhow!("block header with suitable height but no parents"))
-            }
-            (false, Ordering::Greater) => Ok(DagHeaderType::Normal),
-
-            (true, Ordering::Equal) => Ok(DagHeaderType::Genesis),
-            (false, Ordering::Equal) => Err(anyhow!(
-                "block header with dag genesis height but having parents"
-            )),
-
-            (true, Ordering::Less) => Ok(DagHeaderType::Single),
-            (false, Ordering::Less) => Err(anyhow!(
-                "block header with smaller height but having parents"
-            )),
-        }
+    pub fn get_dag_effective_height(&self) -> Result<BlockNumber> {
```

Comment:
The real custom network config would be discarded. It's better not to.

Reply:
I do not know why it is discarded. The logic here is to get a dag height if it is not built in network return 0.

Reply:
Every network has a genesis config, for a custom network with non-zero height, it will be discarded.